torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -639,9 +646,7 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
   virtual void setSequenceNumberForGroup() {
     auto backendType = getBackendType();
     // TODO: HACK for backend name to get sequence number for that backend.
-    if (backendType == ProcessGroup::BackendType::GLOO ||
-        backendType == ProcessGroup::BackendType::NCCL ||
-        backendType == ProcessGroup::BackendType::UCC) {
+    if (backendSupportsSequenceNumbers(backendType)) {
```

Comment:
The current change looks okay, but I don't remember why we'd check the support first. Should we make `setSequenceNumberForGroup` a general method for all backends? (if some backend does not support it, they can throw the `TORCH_CHECK` error below in their impl). 
Cc @H-Huang maybe we can refactor this later.